conda-recipe/build.sh

```diff
@@ -1,41 +1,38 @@
 #!/bin/bash
 
-
 BUILD_CONFIG=Release
 
-if [ `uname` == Linux ]; then
-    CC=${PREFIX}/bin/gcc
-    CXX=${PREFIX}/bin/g++
-
-    # FIXME refactor to reuse the python name (e.g. python3.5m)
-    # FIXME detect any kind of suffix (m, or d)
-    include_path=${PREFIX}/include/python${PY_VER}
-    if [ ! -d $include_path ]; then
-      # Control will enter here if $DIRECTORY doesn't exist.
-      include_path=${PREFIX}/include/python${PY_VER}m
-    fi
-
-    PY_LIB="libpython${PY_VER}.so"
-    PYTHON_LIBRARY=${PREFIX}/lib/${PY_LIB}
-    if [ ! -f $PYTHON_LIBRARY ]; then
-        PYTHON_LIBRARY=${PREFIX}/lib/libpython${PY_VER}m.so
-    fi
-
-    cd pyopcode
-    mkdir build
-    cd build
-
-    cmake ../src \
-        -Wno-dev \
-        -DCMAKE_BUILD_TYPE=${BUILD_CONFIG} \
-        -DCMAKE_INSTALL_PREFIX=${PREFIX} \
-        -DPYTHON_INCLUDE_DIR:PATH=$include_path \
-        -DPYTHON_LIBRARY:FILEPATH=$PYTHON_LIBRARY \
-        -DNUMPY_INCLUDE_DIR:PATH="${SP_DIR}/numpy/core/include" \
-        -DBOOST_ROOT:PATH=${PREFIX}/include
 
+PYTHON_INCLUDE="${PREFIX}/include/python${PY_VER}"
+if [ ! -d ${PYTHON_INCLUDE} ]; then
+    PYTHON_INCLUDE="${PREFIX}/include/python${PY_VER}m"
+fi
+
+PYTHON_LIBRARY_EXT="so"
+if [ `uname` = "Darwin" ] ; then
+    PYTHON_LIBRARY_EXT="dylib"
+fi
+
+PYTHON_LIBRARY="${PREFIX}/lib/libpython${PY_VER}.${PYTHON_LIBRARY_EXT}"
+if [ ! -f ${PYTHON_LIBRARY} ]; then
+    PYTHON_LIBRARY="${PREFIX}/lib/libpython${PY_VER}m.${PYTHON_LIBRARY_EXT}"
 fi
 
+cd pyopcode
+mkdir build
+cd build
+
+cmake ../src \
+    -Wno-dev \
+    -DCMAKE_BUILD_TYPE=${BUILD_CONFIG} \
+    -DCMAKE_CXX_FLAGS="-Wreturn-type" \
+    -DCMAKE_OSX_DEPLOYMENT_TARGET="10.12" \
```

Comment:
Conda-build provides some env vars to maintain compatibility throughout the Anaconda ecosystem: http://conda.pydata.org/docs/building/environment-vars.html

In this case I think you should use `MACOSX_DEPLOYMENT_TARGET`; and be careful not to break the linux build where this envvar is unavailable :)